core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -101,9 +104,11 @@ public void filter(ContainerRequestContext request)
                 return;
             }
             String principalName = (String) principal;
-            setAuthenticatedIdentity(request, Identity.forUser(userMapping.mapUser(principalName))
-                    .withPrincipal(new BasicPrincipal(principalName))
-                    .build());
+            Identity.Builder identity = Identity.forUser(userMapping.mapUser(principalName));
+            identity.withPrincipal(new BasicPrincipal(principalName));
+            groupsField.flatMap(field -> Optional.ofNullable((List<String>) claims.get().get(field)))
```

Comment:
Again `map` here is cleaner and clearer.

Reply:
userMapping.mapUser(principalName) throws checked UserMappingException